.github/workflows/publish.yml

```diff
@@ -0,0 +1,87 @@
+name: publish artifacts
+on:
+  push:
+    branches:
+      - 'main'
```

Comment:
when drafting an action, leave this push trigger out and test it out with manual workflow-dispatch calls on a branch